cmd/s3-peer-bms-client.go

```diff
@@ -0,0 +1,149 @@
+/*
+ * Minio Cloud Storage, (C) 2014-2016 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+)
+
+// Bucket Meta State interface to be registered in S3 Peers API
+type s3PeerBucketMetaStateAPI struct{}
+
+// Bucket Meta State API name
+func (c s3PeerBucketMetaStateAPI) APIName() string {
+	return "bms"
+}
+
+// Return Bucket Meta State API for local use
+func (c s3PeerBucketMetaStateAPI) NewLocalClient() interface{} {
```

Comment:
Actually this PR will help to implement new calls (local and remote) like Get Storage Info (https://github.com/minio/minio/issues/3475).. that's why I am using generic interface{}